packages/fxa-content-server/app/scripts/lib/experiments/grouping-rules/signup-code.js

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const BaseGroupingRule = require('./base');
+const Constants = require('../../../lib/constants');
+const GROUPS_DEFAULT = ['control, treatment'];
+
+const ROLLOUT_CLIENTS = {
+  '37fdfa37698f251a': {
+    enableTestEmails: false,
+    groups: GROUPS_DEFAULT,
+    name: 'Lockbox Extension',
+    rolloutRate: 0.0,
+  },
+  '3c49430b43dfba77': {
+    enableTestEmails: false,
+    groups: GROUPS_DEFAULT,
+    name: 'Android Components Reference Browser',
+    rolloutRate: 1.0,
+  },
+  '98adfa37698f255b': {
+    enableTestEmails: true,
+    groups: GROUPS_DEFAULT,
+    name: 'Lockbox Extension iOS',
+    rolloutRate: 0.0,
+  },
+  ecdb5ae7add825d4: {
+    enableTestEmails: false,
+    groups: GROUPS_DEFAULT,
+    name: 'TestClient',
+    rolloutRate: 0.0,
+  },
+};
+
+module.exports = class SignupCodeGroupingRule extends BaseGroupingRule {
+  constructor() {
+    super();
+    this.name = 'signupCode';
+    this.SYNC_ROLLOUT_RATE = 0.0;
+    this.ROLLOUT_CLIENTS = ROLLOUT_CLIENTS;
+  }
+
+  choose(subject) {
+    if (
+      !subject ||
+      !subject.uniqueUserId ||
+      !subject.experimentGroupingRules ||
+      !subject.isSignupCodeSupported ||
+      !subject.account
+    ) {
+      return false;
+    }
+
+    const { featureFlags } = subject;
+
+    if (subject.clientId) {
+      let client = this.ROLLOUT_CLIENTS[subject.clientId];
+      if (featureFlags && featureFlags.signupCodeClients) {
+        client = featureFlags.signupCodeClients[subject.clientId];
```

Comment:
Great! If/when we want to tweak the rollout rates we can do that through Redis, you'll be the guinea pig!